src/services/autoQueueService.ts

```diff
@@ -0,0 +1,41 @@
+import {
+  useQueueSettingsStore,
+  useQueuePendingTaskCountStore
+} from '@/stores/queueStore'
+import { app } from '@/scripts/app'
+import { api } from '@/scripts/api'
+
+export function setupAutoQueueHandler() {
+  const queueCountStore = useQueuePendingTaskCountStore()
+  const queueSettingsStore = useQueueSettingsStore()
+
+  let graphHasChanged = false
+  let internalCount = 0 // Use an internal counter here so it is instantly updated when re-queuing
```

Comment:
Can you further explain how this `internalCount` works? Is it to force `queuePrompt` when graph changed? I think the code here needs some documentation / testing, as it is hard to understand.

Some suggestions on how to make this more readable:
- We can create a mini-store that stores the graph version counter that increments every time the graph changes 
- Register watch for the counter if we are in change mode, the watch callback call app.queuePrompt everytime there is a graph change
- Watch for `queueCountStore.count` if we are in instant mode, the watch callback call app.queuePrompt everytime the pending task reaches 0.

I think doing 2 watch function makes the logic a lot cleaner as here the internalCount is used for both the graph version and the pending task count.

Reply:
`internalCount` is just the current store count value, or that +1 directly after an automatic queue. If you don't have the manual increment then you'll end up with a case where you can queue multiple changes before the store count value is updated.
(store=0, make change, auto queue prompt, make change, this auto queues again, websocket received, store=2)
I don't think that your suggestion will help and will still require this. 
This mechanism wouldn't want this to update the store value as if the graph is invalid and rejected by the server then the UI would show 1 thing in the queue and not update until the next event.

Reply:
If that is the case, can we have some playwright tests to verify the fast click of queue button works as intended? You can add some dummy execution nodes in https://github.com/Comfy-Org/ComfyUI_devtools to mock the execution.

Reply:
Done